src/stream/src/executor_v2/mod.rs

```diff
@@ -129,3 +156,309 @@ pub trait Executor: Send + 'static {
         Box::new(self)
     }
 }
+
+pub const INVALID_EPOCH: u64 = 0;
+
+pub trait ExprFn = Fn(&DataChunk) -> Result<Bitmap> + Send + Sync + 'static;
+
+/// Boxed stream of [`StreamMessage`].
+pub type BoxedExecutorV1Stream = Pin<Box<dyn Stream<Item = Result<Message>> + Send>>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Mutation {
+    Stop(HashSet<ActorId>),
+    UpdateOutputs(HashMap<(ActorId, DispatcherId), Vec<ActorInfo>>),
+    AddOutput(HashMap<(ActorId, DispatcherId), Vec<ActorInfo>>),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Epoch {
+    pub curr: u64,
+    pub prev: u64,
+}
+
+impl Epoch {
+    pub fn new(curr: u64, prev: u64) -> Self {
+        assert!(curr > prev);
+        Self { curr, prev }
+    }
+
+    pub fn inc(&self) -> Self {
```

Comment:
Should rename it to something like `inc_in_test()` or mark it as `cfg(test)`. Epochs don't have such properties generally.